### ISSUE TITLE
Using pkg-config to find libxml2

### ISSUE THREAD

**Issue by hughmcmaster** [vlvassilev/yuma123#129]
Debian is taking steps to remove xml2-config in favour of pkg-config. This means yuma123 will fail to build from source.

Making the switch to pkg-config is simple, although there are some design decisions to be made.

1. Replacing `AM_PATH_XML2` with `PKG_CHECK_MODULES([XML], [libxml-2.0])`. This would need to done across all configure.ac files in your source, including those in the module examples. You could also retain support for `XML2_CONFIG` if desired.
2. As PKG_CHECK_MODULES will automatically define (in this case) the variables XML_CFLAGS and XML_LIBS, it would be nice to use them directly. Your package uses XML_CPPFLAGS and XML_LIBS. While we can easily add `AC_SUBST([XML_CPPFLAGS], [$XML_CFLAGS])` to each configure.ac file, I think it'd be simpler and more logical to update the various Makefile.am files to reference $XML_CFLAGS instead.

I'm happy to take care of these changes. Please let me know your preferences on the above points. Otherwise, let me know how you want to proceed.

**Comment by vlvassilev** [comment]
Thanks for taking initiative on this. How about this (seems to work):

diff --git a/configure.ac b/configure.ac
index 36e75185..f91e31f1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -84,7 +84,14 @@ AC_ARG_WITH(tecla,
 AM_CONDITIONAL([WITH_TECLA], [test "x$TECLA" = x1])
 AM_CONDITIONAL([STANDALONE], [test x$standalone = xtrue])
 
-AM_PATH_XML2
+PKG_CHECK_MODULES(LIBXML2, libxml-2.0 >= 2.3.9)
+XML_LIBS=$LIBXML2_LIBS
+XML_CFLAGS=$LIBXML2_CFLAGS
+XML_CPPFLAGS=$LIBXML2_CFLAGS
+AC_SUBST(XML_LIBS)
+AC_SUBST(XML_CFLAGS)
+AC_SUBST(XML_CPPFLAGS)
+
 
 AC_CONFIG_FILES([
         Makefile \

**Comment by hughmcmaster** [comment]
That will definitely work, although you can simplify the changes quite a bit.

By making the variable prefix `XML`, no changes are required for your existing `$(LIB_XML)` variable references. And `AC_SUBST` is called automatically by `PKG_CHECK_MODULES`.

You also don't need XML_CFLAGS, since it's not used in the source at all.

Of course, if you want to use `LIBXML2` as the variable prefix, you can just drop the `XML_CFLAGS` lines. But I'd recommend quoting with square brackets to avoid any issues.

```
 diff --git a/configure.ac b/configure.ac
 index 36e75185..4485e494 100644
 --- a/configure.ac
 +++ b/configure.ac
 @@ -12,6 +12,8 @@ AC_SEARCH_LIBS([dlopen], [dl dld], [], [
    AC_MSG_ERROR([unable to find the dlopen() function])
  ])
  
 +PKG_PROG_PKG_CONFIG
 +
  netconfmoduledir="$libdir/yuma"
  AC_SUBST(netconfmoduledir)
  
 @@ -84,7 +86,9 @@ AC_ARG_WITH(tecla,
  AM_CONDITIONAL([WITH_TECLA], [test "x$TECLA" = x1])
  AM_CONDITIONAL([STANDALONE], [test x$standalone = xtrue])
  
 -AM_PATH_XML2
 +PKG_CHECK_MODULES([XML], [libxml-2.0 >= 2.3.9])
 +XML_CPPFLAGS="$XML_CFLAGS"
 +AC_SUBST([XML_CPPFLAGS])
```

**Comment by hughmcmaster** [comment]
Fixed by #130, #131 and #132. Closing.

**Comment by vlvassilev** [comment]
@hughmcmaster i always thought enforcing the mandatory requirement of including  libxml/xmlstring.h in all applications was not justified.

This is done mostly because of the xmlChar type. There are very few applications that actually make use of the low level libxml2 helper routines. So the majority of the examples and  tests must have the libxml2 development files without a good reason.

He is my take in what could be done to avoid that dependency without changing all of the thousands of instances of xmlChar: vlvassilev/yuma123@80a0283173d811af2b46e2951a0ef72375fffeba
